packages/js/email-editor/src/components/preview/preview-save-guard.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useCallback, useEffect } from '@wordpress/element';
+import { select, dispatch } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * A component that adds a confirmation dialog for unsaved changes
+ * when a user tries to open a preview in a new tab.
+ */
+export const PreviewSaveGuard = () => {
+	const selector = '.editor-preview-dropdown__button-external';
+
+	/**
+	 * Handles click/keydown events to check for unsaved changes before previewing.
+	 *
+	 * @param {Event} event The triggered event.
+	 */
+	const guard = async ( event ) => {
+		const target = event.target;
+
+		// Find the triggering element (the preview button).
+		const triggerEl = target?.closest( selector );
+		if ( ! triggerEl ) {
+			return;
+		}
+
+		// Check if there are any unsaved changes using the editor data store.
+		const editorStoreInstance = select( editorStore );
+		const isDirty = editorStoreInstance?.isEditedPostDirty?.();
+
+		if ( ! isDirty ) {
+			// If the post is saved, do nothing and let the preview open normally.
+			return;
+		}
+
+		// If there are unsaved changes, prevent the default action (opening the link).
+		event.preventDefault();
+		event.stopPropagation();
+		event.stopImmediatePropagation();
+
+		// Dispatch a notice to inform the user they need to save first.
+		dispatch( noticesStore ).createNotice(
+			'info',
```

Comment:
Is this notice visible enough? I feer a bit that the user may overlook it and feel like the preview functionality is broken.

Reply:
This was in the ticket spec. We can use a regular notice if you want. I don't know what is better here.
<img width="1889" height="494" alt="Screenshot 2025-08-14 at 17 57 31" src="https://github.com/user-attachments/assets/a381188c-629d-4fb4-9a23-1cedbf934a46" />